src/fenicsxconcrete/finite_element_problem/concrete_am.py

```diff
@@ -0,0 +1,506 @@
+
+        # quadrature functions (required in pde)
+        self.q_E = df.fem.Function(q_V, name="youngs_modulus")
+        self.q_fd = df.fem.Function(q_V, name="density_increment")
+
+        # path variable from AM Problem
+        self.q_array_path = self.rule.create_quadrature_array(self.mesh, shape=1)
+        self.q_array_path[:] = 0.0  # default all active set by "set_initial_path(q_array_path)"
+        # pseudo density for element activation
+        self.q_array_pd = self.rule.create_quadrature_array(self.mesh, shape=1)
+
+        self.q_sig = df.fem.Function(q_VT, name="stress")
+        self.q_eps = df.fem.Function(q_VT, name="strain")
+
+        # standard space
+        self.V = u.function_space
+
+        # Define variational problem
+        v = ufl.TestFunction(self.V)
+
+        # build up form
+        # multiplication with activated elements / current Young's modulus
+        R_ufl = ufl.inner(self.sigma(u), self.epsilon(v)) * self.rule.dx
+
+        # apply body force
+        body_force = body_force_fct(v, self.q_fd, self.rule)
+        if body_force:
+            R_ufl -= body_force
+
+        # quadrature point part
+        self.R = R_ufl
+
+        # derivative
+        # normal form
+        dR_ufl = ufl.derivative(R_ufl, u)
+
+        # quadrature part
+        self.dR = dR_ufl
+        self.sigma_evaluator = QuadratureEvaluator(self.sigma(u), self.mesh, self.rule)
+        self.eps_evaluator = QuadratureEvaluator(self.epsilon(u), self.mesh, self.rule)
+
+        super().__init__(self.R, u, bc, self.dR)
+
+    def x_sigma(self, v: ufl.core.expr.Expr) -> ufl.core.expr.Expr:
+        """compute stresses for Young's modulus == 1
+
+        Args:
+            v: testfunction
+
+        Returns:
+            ufl expression for sigma
+        """
+
+        x_mu = df.fem.Constant(self.mesh, 1.0 / (2.0 * (1.0 + self.p["nu"])))
+        x_lambda = df.fem.Constant(self.mesh, 1.0 * self.p["nu"] / ((1.0 + self.p["nu"]) * (1.0 - 2.0 * self.p["nu"])))
+        if self.p["dim"] == 2 and self.p["stress_state"] == "plane_stress":
+            # see https://comet-fenics.readthedocs.io/en/latest/demo/elasticity/2D_elasticity.py.html
+            x_lambda = df.fem.Constant(self.mesh, 2 * x_mu.value * x_lambda.value / (x_lambda.value + 2 * x_mu.value))
+
+        return 2.0 * x_mu * self.epsilon(v) + x_lambda * ufl.nabla_div(v) * ufl.Identity(self.p["dim"])
+
+    def sigma(self, v: ufl.core.expr.Expr) -> ufl.core.expr.Expr:
+        """computes stresses for real Young's modulus given as quadrature fct q_E
+
+        Args:
+            v: testfunction
+
+        Returns:
+            ufl expression for sigma
+        """
+
+        return self.q_E * self.x_sigma(v)
+
+    def epsilon(self, v: ufl.core.expr.Expr) -> ufl.core.expr.Expr:
+        """computes strains
+
+        Args:
+            v: testfunction
+
+        Returns:
+            ufl expression for strain
+        """
+        return ufl.tensoralgebra.Sym(ufl.grad(v))
```

Comment:
Is there a reason you used this line instead of `ufl.sym(ufl.grad(v))`? Or is it the same?
```suggestion
        return ufl.sym(ufl.grad(v))
```